api/secret.go

```diff
@@ -58,11 +64,16 @@ type SecretAuth struct {
 
 // ParseSecret is used to parse a secret value from JSON from an io.Reader.
 func ParseSecret(r io.Reader) (*Secret, error) {
+	var rawBuffer bytes.Buffer
+	secretReader := io.TeeReader(r, &rawBuffer)
+
 	// First decode the JSON into a map[string]interface{}
 	var secret Secret
-	if err := jsonutil.DecodeJSONFromReader(r, &secret); err != nil {
+	if err := jsonutil.DecodeJSONFromReader(secretReader, &secret); err != nil {
 		return nil, err
 	}
 
+	secret.RawSecret = rawBuffer.Bytes()
```

Comment:
Thoughts on actually just returning a `*bytes.Buffer` instead? That way if people want to read the bytes they have something that implements `Reader` ready to go, and other useful things.